src/s3/utils.rs

```diff
@@ -68,17 +69,15 @@ pub fn uint32(mut data: &[u8]) -> Result<u32, std::io::Error> {
 
 /// Gets hex encoded SHA256 hash of given data
 pub fn sha256_hash(data: &[u8]) -> String {
-    let mut hasher = Sha256::new();
-    hasher.update(data);
-    format!("{:x}", hasher.finalize())
+    ring::digest::digest(&SHA256, data).encode_hex()
 }
 
 pub fn sha256_hash_sb(sb: &SegmentedBytes) -> String {
-    let mut hasher = Sha256::new();
+    let mut context = Context::new(&SHA256);
     for data in sb.iter() {
-        hasher.update(data);
+        context.update(data.as_ref());
     }
-    format!("{:x}", hasher.finalize())
```

Comment:
Also AFAIK this is not a great way to convert to hex string. I personally avoid `format!` as much as possible and in this particular case, I would either use:
- `hex::encode` since we already have it as a dependency.
- `const_hex::encode` new shiny kid that is 10x~50x faster than `hex`.

Reply:
Correct, this was in upstream and I used hex crate to encode in my patch since it was already a dependency. TIL about const_hex, will check it out. Thanks!